src/detect-content.c

```diff
@@ -379,20 +379,29 @@ _Bool DetectContentPMATCHValidateCallback(const Signature *s)
     if (!(s->flags & SIG_FLAG_DSIZE)) {
         return TRUE;
     }
-
     int max_right_edge_i = SigParseGetMaxDsize(s);
     if (max_right_edge_i < 0) {
         return TRUE;
     }
 
     uint32_t max_right_edge = (uint32_t)max_right_edge_i;
+    int content_total_length = 0;
```

Comment:
nit: with the way this is implemented, neither `content_total_length` nor `previous_content_length` require initialization values.

Reply:
Okay, I will update in the next version. Thanks Jeff! For my own information, is it just style or functionally bad?

Reply:
It's not harmful but initialization in this case is not necessary either. So, it's a (minor) optimization to avoid doing unnecessary work. But, it is minor. Best to include only what's strictly necessary, IMO.

Reply:
Cool, thanks!

Reply:
I think ```content_total_length``` can be declared on first use, which is within the for loop below.

Reply:
Looks like this will neatly apply to all three vars?